src/lib/oauth.ts

```diff
@@ -77,16 +78,17 @@ export async function handleAuthCallback( hash: string ): Promise< Error | Store
 	if ( isNaN( expiresIn ) || expiresIn === 0 || ! accessToken ) {
 		return new Error( 'Error while getting token' );
 	}
-	let response: { email?: string; display_name?: string } = {};
+	let response: { ID?: number; email?: string; display_name?: string } = {};
 	try {
-		response = await new wpcom( accessToken ).req.get( '/me?fields=email,display_name' );
+		response = await new wpcom( accessToken ).req.get( '/me?fields=ID,email,display_name' );
 	} catch ( error ) {
 		Sentry.captureException( error );
 	}
 	return {
 		expiresIn,
 		expirationTime: new Date().getTime() + expiresIn * 1000,
 		accessToken,
+		id: response.ID,
 		email: response.email,
 		displayName: response.display_name,
 	};
```

Comment:
I confirm the user user.id appears in the `~Library/Application Support/Studio/appdata-v1.json`

<img width="186" alt="Screenshot 2024-06-20 at 13 30 29" src="https://github.com/Automattic/studio/assets/779993/e363dd28-d21c-48a8-8b2a-120bd5e7ebea">